src/driver/amdxdna/amdxdna_mailbox.c

```diff
@@ -490,15 +507,14 @@ static void mailbox_polld_handle_chann(struct mailbox_channel *mb_chann)
 	if (mb_chann->bad_state)
 		return;
 
-	iohub = mailbox_reg_read(mb_chann, mb_chann->iohub_int_addr);
+	iohub = mailbox_irq_status(mb_chann);
 	if (!iohub)
```

Comment:
This change doesn't work with the mb_chann->iohub_int_addr == NULL code path. The iohub will always be 0.
This function will always return without doing anything. It makes turbo mode not working when mb_chann->iohub_int_addr == NULL.